app/APIs/NeedAuthAPI.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: macintosh
+ * Date: 8/5/15
+ * Time: 5:16 PM
+ */
+
+namespace App\APIs;
+
+
+use App\Libs\ColdValidator;
+use App\ModelFactory\UserSessionFactory;
+use App\Models\LoginSession;
+use App\Models\Device;
+use App\Models\User;
+
+class NeedAuthAPI extends BaseAPI
+{
+    const INVALID_SESS_KEY = 101;
+    private static $errorMsg = array(
+        self::INVALID_SESS_KEY => 'Invalid session key'
+    );
+
+    public $isPublic = true;
+    public $dontNeedAuth = false;
+
+    protected $currentSession;
+
+    public function __construct() {
+        if($this->dontNeedAuth == true) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1125)